examples/experimental/dagster-dlift/dagster_dlift/env_client.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, Mapping, Sequence, Type, Union
+
+from dagster import AssetCheckResult, AssetMaterialization
+from dagster._record import record
+
+from dagster_dlift.client import DbtCloudClient
+from dagster_dlift.translator import (
+    DagsterDbtCloudRuntimeTranslator,
+    DagsterDbtCloudTranslator,
+    DbtCloudProjectEnvironmentData,
+)
+
+
+@record
+class DbtCloudJobRun:
+    """Represents a dbt Cloud job run."""
+
+    job_id: int
+    run_id: int
+    client: DbtCloudClient
+    translator: DagsterDbtCloudTranslator
+    env_data: DbtCloudProjectEnvironmentData
+
+    def wait_for_success(self) -> int:
+        return self.client.poll_for_run_completion(self.run_id)
+
+    def get_run_results(self) -> Mapping[str, Any]:
+        return self.client.get_run_results_json(self.run_id)
+
+    def get_asset_events(
```

Comment:
Should this be an iterator so that we can yield events as they come?

Reply:
good idea.